src/hyperon_das_query_engine/hyperon_das_query_engine_ext.cc

```diff
@@ -0,0 +1,85 @@
+#include <nanobind/nanobind.h>
+#include <nanobind/stl/string.h>
+//TODO: remove unused imports
+// #include <nanobind/stl/vector.h>
+// #include <nanobind/stl/shared_ptr.h>
+#include <nanobind/trampoline.h>
+#include <type_traits>
+
+#include "query_engine/DASNode.h"
+#include "query_engine/HandlesAnswer.h"
+#include "query_engine/QueryAnswer.h"
+
+#include "query_engine/query_element/RemoteIterator.h"
+
+namespace nb = nanobind;
+using namespace nb::literals;
+
+using namespace std;
+using namespace query_engine;
+
+NB_MODULE(hyperon_das_query_engine_ext, m) {
+  //QueryAnswer.h bindings
+  nb::class_<QueryAnswer>(m, "QueryAnswer")
+    .def("tokenize", &QueryAnswer::tokenize)
+    .def("untokenize", &QueryAnswer::untokenize, "tokens"_a)
+    .def("to_string", &QueryAnswer::to_string);
+
+  //HandlesAnswer.h
+  nb::class_<HandlesAnswer, QueryAnswer>(m, "HandlesAnswer")
+    .def(nb::init<>())
+    .def(nb::init<double>(), "importance"_a)
+    .def(nb::init<const char*, double>(), "handle"_a, "importance"_a)
+    //TODO: nanobind is failing to bind the handles field
+    // error: invalid conversion from 'const char* const*' to 'char'
+    // .def_ro("handles", &HandlesAnswer::handles)
+    .def_ro("handles_size", &HandlesAnswer::handles_size)
+    .def_ro("importance", &HandlesAnswer::importance)
+    .def("add_handle", &HandlesAnswer::add_handle, "handle"_a)
+    .def("merge", &HandlesAnswer::merge, "other"_a, "merge_handles"_a)
+    .def_static("copy", &HandlesAnswer::copy, "base"_a)
+  ;
+
+  // RemoteIterator.h
+  nb::class_<RemoteIterator<HandlesAnswer>>(m, "RemoteIterator")
+   .def(nb::init<string>(), "local_id"_a)
```

Comment:
```suggestion
   .def(nb::init<const string&>(), "local_id"_a)
```

Please revisit the other functions/constructors to ensure that the binding is expecting/passing in the right types, including aspect modifiers like `const`, `&`, etc.

Reply:
Done: [`18c72c0` (#299)](https://github.com/singnet/das/pull/299/commits/18c72c0e64e74cf01f7ae71d068039e1b81d3487)